src/mongo_odm/nvp.hpp

```diff
@@ -21,6 +21,24 @@
 #include <tuple>
 #include <type_traits>
 
+#include <bsoncxx/types.hpp>
+
+#include <mongo_odm/util.hpp>
+
+// Preprocessor templates for manipulating multiple arguments.
+#define PP_NARG(...) PP_NARG_(__VA_ARGS__, PP_RSEQ_N())
```

Comment:
Actually, let's drag them out into their own file and namespace the macros.

Nothing about the implementation requires them to be in this particular header